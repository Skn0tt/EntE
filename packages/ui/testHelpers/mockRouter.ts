import { RouteComponentProps } from "react-router";
import { Location, UnregisterCallback } from "history";

export function getMockRouterProps<P>(data: P) {
  const location: Location = {
    hash: "",
    key: "",
    pathname: "",
    search: "",
    state: {}
  };

  const props: RouteComponentProps<P> = {
    location,
    match: {
      isExact: true,
      params: data,
      path: "",
      url: ""
    },
    history: {
      location,
      length: 2,
      action: "POP",
      push: () => {},
      replace: () => {},
      go: num => {},
      goBack: () => {},
      goForward: () => {},
      block: t => null as any,
      createHref: t => "",
      listen: t => null as any
    },
    staticContext: {}
  };

  return props;
}
