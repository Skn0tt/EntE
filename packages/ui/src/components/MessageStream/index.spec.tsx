import * as React from "react";
import { MessageStream } from "./index";
import { shallow } from "enzyme";
import { getMockRouterProps } from "../../../test_helpers/mockRouter";
import * as ReactRouterEnzymeContext from "react-router-enzyme-context";

describe("MessageStream", () => {
  const removeMessage = jest.fn();
  const messages = ["Hallo", "Nachricht"];

  const comp = shallow(
    <MessageStream
      messages={messages}
      removeMessage={removeMessage}
      classes={{}}
    />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });
});
