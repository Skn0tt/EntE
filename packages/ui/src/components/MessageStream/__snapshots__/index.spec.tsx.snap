// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageStream renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    classes={Object {}}
    messages={
      Array [
        "Hallo",
        "Nachricht",
      ]
    }
    removeMessage={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <WithStyles(Snackbar)
          action={
            <WithStyles(IconButton)
              className="remove0"
              color="inherit"
              onClick={[Function]}
            >
              <pure(Close) />
            </WithStyles(IconButton)>
          }
          autoHideDuration={6000}
          message={
            <span>
              Hallo
            </span>
          }
          onClose={[Function]}
          open={true}
        />,
        <WithStyles(Snackbar)
          action={
            <WithStyles(IconButton)
              className="remove1"
              color="inherit"
              onClick={[Function]}
            >
              <pure(Close) />
            </WithStyles(IconButton)>
          }
          autoHideDuration={6000}
          message={
            <span>
              Nachricht
            </span>
          }
          onClose={[Function]}
          open={true}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "action": <WithStyles(IconButton)
            className="remove0"
            color="inherit"
            onClick={[Function]}
          >
            <pure(Close) />
          </WithStyles(IconButton)>,
          "autoHideDuration": 6000,
          "message": <span>
            Hallo
          </span>,
          "onClose": [Function],
          "open": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "action": <WithStyles(IconButton)
            className="remove1"
            color="inherit"
            onClick={[Function]}
          >
            <pure(Close) />
          </WithStyles(IconButton)>,
          "autoHideDuration": 6000,
          "message": <span>
            Nachricht
          </span>,
          "onClose": [Function],
          "open": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <WithStyles(Snackbar)
            action={
              <WithStyles(IconButton)
                className="remove0"
                color="inherit"
                onClick={[Function]}
              >
                <pure(Close) />
              </WithStyles(IconButton)>
            }
            autoHideDuration={6000}
            message={
              <span>
                Hallo
              </span>
            }
            onClose={[Function]}
            open={true}
          />,
          <WithStyles(Snackbar)
            action={
              <WithStyles(IconButton)
                className="remove1"
                color="inherit"
                onClick={[Function]}
              >
                <pure(Close) />
              </WithStyles(IconButton)>
            }
            autoHideDuration={6000}
            message={
              <span>
                Nachricht
              </span>
            }
            onClose={[Function]}
            open={true}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "action": <WithStyles(IconButton)
              className="remove0"
              color="inherit"
              onClick={[Function]}
            >
              <pure(Close) />
            </WithStyles(IconButton)>,
            "autoHideDuration": 6000,
            "message": <span>
              Hallo
            </span>,
            "onClose": [Function],
            "open": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "action": <WithStyles(IconButton)
              className="remove1"
              color="inherit"
              onClick={[Function]}
            >
              <pure(Close) />
            </WithStyles(IconButton)>,
            "autoHideDuration": 6000,
            "message": <span>
              Nachricht
            </span>,
            "onClose": [Function],
            "open": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
